### ISSUE TITLE
Fix Trie not found root hash bug when running Bacteria

### ISSUE THREAD

**Issue by ilblackdragon** [nearprotocol/nearcore#304]
Please use either storage from the server to reproduce or reproduce by running bacteria.

**Comment by ilblackdragon** [comment]
Here is repro test (includes bacteria.wasm from http://studio.nearprotocol.com/?f=JvaTmxP):
```
    #[test]
    fn test_trie_failure() {
        let (mut runtime, viewer) = get_runtime_and_state_db_viewer();
        let mut root = viewer.get_root();
        let pub_key = "DWsBaCSq9BK9CKnEynDmfXwoziFJ3UwUZCN2ye5FHZFa";
        // let (pub_key, _) = get_keypair();
        let wasm_binary = include_bytes!("../../../tests/bacteria.wasm");
        let tx_body = TransactionBody::DeployContract(DeployContractTransaction {
            nonce: 1,
            sender: hash(b"alice"),
            contract_id: hash(b"studio-JvaTmxP"),
            public_key: PublicKey::from(pub_key).encode().unwrap(),//pub_key.encode().unwrap(),
            wasm_byte_array: wasm_binary.to_vec(),
        });
        println!("Genesis {}", root);
        root = apply_transaction(&mut runtime, root, tx_body);
        println!("Deploy contract {}", root);
        let init_state_tx = "{\"FunctionCall\":{\"nonce\":1,\"originator\":[129,182,55,216,252,210,198,218,99,89,230,150,49,19,161,23,13,231,149,228,183,37,184,77,30,11,76,253,158,197,140,233],\"contract_id\":[131,129,26,91,252,192,78,246,253,210,17,154,124,82,20,63,176,26,163,159,165,166,38,119,130,34,2,176,185,188,1,249],\"method_name\":[110,101,97,114,95,102,117,110,99,95,105,110,105,116,83,116,97,116,101],\"args\":[69,0,0,0,2,115,116,97,116,101,95,105,100,0,8,0,0,0,115,116,97,116,101,49,50,0,2,98,111,116,49,0,11,0,0,0,51,51,51,51,51,51,51,51,51,51,0,2,98,111,116,50,0,11,0,0,0,51,51,51,51,51,51,51,51,51,51,0,0],\"amount\":0}}";
        let tx_body: TransactionBody = serde_json::from_str(init_state_tx).unwrap();
        root = apply_transaction(&mut runtime, root, tx_body);
        println!("Init state {}", root);
        for nonce in 2..32 {
            let tx_body: TransactionBody = serde_json::from_str(
                &format!("{{\"FunctionCall\":{{\"nonce\":{},\"originator\":[129,182,55,216,252,210,198,218,99,89,230,150,49,19,161,23,13,231,149,228,183,37,184,77,30,11,76,253,158,197,140,233],\"contract_id\":[131,129,26,91,252,192,78,246,253,210,17,154,124,82,20,63,176,26,163,159,165,166,38,119,130,34,2,176,185,188,1,249],\"method_name\":[110,101,97,114,95,102,117,110,99,95,97,100,118,97,110,99,101,83,116,97,116,101],\"args\":[27,0,0,0,2,115,116,97,116,101,95,105,100,0,8,0,0,0,115,116,97,116,101,49,50,0,0],\"amount\":0}}}}", nonce)).unwrap();
            root = apply_transaction(&mut runtime, root, tx_body);
            println!("Root {}", root);
        }
    }
```

**Comment by ilblackdragon** [comment]
On the last step, TrieDbMut creates a transaction that removes all the keys from underlaying db. Continue investigating.

**Comment by ilblackdragon** [comment]
Fixed via #333
